nixos/modules/services/databases/postgresql.nix

```diff
@@ -532,7 +622,67 @@ in
           '';
 
         # Wait for PostgreSQL to be ready to accept connections.
-        postStart =
+        postStart = let
+          createDatabase = ps: let
+            addS = n: p: optional (p != null) "${n} '${p}'";
+            addL = n: p: optional (p != null) "${n} ${toString p}";
+          in
+            concatStringsSep " " (
+              [ "CREATE DATABASE \"${ps.name}\"" ]
+              ++ addS "OWNER" ps.owner
+              ++ addS "TEMPLATE" ps.template
+              ++ addS "ENCODING" ps.encoding
+              ++ addS "LC_COLLATE" ps.lcCollate
+              ++ addS "LC_CTYPE" ps.lcCtype
+              ++ addL "ALLOW_CONNECTIONS" ps.allowConnections
+              ++ addL "CONNECTION LIMIT" ps.connectionLimit
+              ++ addL "IS_TEMPLATE" ps.isTemplate);
+
+          # Run a single SQL command using psql with a list of additional
+          # command line options.
+          psqlCmd = cmd: args: concatStringsSep " " (
+            [ "$PSQL -tA" ]
+            ++ args
```

Comment:
`args` doesn't seem to be used. What do you think of removing it?

Reply:
`args` is actually used a few lines below in `ensureUsers`